test/document_stores/test_faiss.py

```diff
@@ -204,6 +204,19 @@ def test_update_docs_different_indexes(self, ds, documents_with_embeddings):
         assert len(docs_from_index_b) == len(docs_b)
         assert {int(doc.meta["vector_id"]) for doc in docs_from_index_b} == {0, 1, 2, 3}
 
+    @pytest.mark.integration
+    def test_dont_update_existing_embeddings(self, ds, docs):
+        retriever = MockDenseRetriever(document_store=ds)
+        first_doc_id = docs[0].id
+
+        for i in range(1, 4):
+            ds.write_documents(docs[:i])
+            ds.update_embeddings(retriever=retriever, update_existing_embeddings=False)
+
+            assert ds.get_document_count() == i
+            assert ds.get_embedding_count() == i
+            assert ds.get_document_by_id(id=first_doc_id).meta["vector_id"] == "0"
```

Comment:
Nitpick: can we check that also the embedding didn't change? Otherwise this test might be passing in the case where `update_existing_embeddings` gets ignored by `update_embeddings` (for whatever reason).

Reply:
Makes sense!